sagemaker-clarify/fairness_and_explainability/fairness_and_explainability_boto3.ipynb

```diff
@@ -0,0 +1,1541 @@
+      "}"
+     ]
+    }
+   ],
+   "source": [
+    "!echo\n",
+    "!cat analysis_config/bias_analysis_config.json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`bias_analysis_config.json` here contains configuration values for detecting bias using a clarify container:\n",
+    "* `dataset_type` specifies the format of your dataset, for thsi example as we are using CSV dataset this will be `text/csv`\n",
+    "* `headers` is the list of column names in the dataset\n",
+    "* `label` specifies the ground truth label, which is also known as observed label or target attribute. It is used for many bias metrics. In this example, the \"Target\" column has the ground truth label.\n",
+    "* `facet`: SageMaker Clarify also needs information on what the sensitive columns (`facets`) are, what the sensitive features (`facet: values_or_threshold`) may be, and what the desirable outcomes are (`label_values_or_threshold`).\n",
+    "SageMaker Clarify can handle both categorical and continuous data for `facet: values_or_threshold` and for `label_values_or_threshold`. In this case we are using categorical data. This example has selected the sensitive attribute `Sex` to analyze. The results will show if the model has a preference for records of one sex over the other.\n",
+    "* `group_variable`: This example has selected the \"Age\" column which is used to form subgroups for the measurement of bias metric [Conditional Demographic Disparity (CDD)](https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-data-bias-metric-cddl.html) or [Conditional Demographic Disparity in Predicted Labels (CDDPL)](https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-post-training-bias-metric-cddpl.html).\n",
+    "* `probability_threshold` is to indicate the threshold to select the binary label in the case of binary classification. XGBoost model outputs probabilities of samples, so SageMaker Clarify invokes the endpoint then uses `probability_threshold` to convert the probability to binary labels for bias analysis. Prediction above the threshold is interpreted as label value `1` and below or equal as label value `0`.\n",
+    "* `methods` is the list of methods and their parameters for the analyses and reports. If any section is omitted, then it is not computed.\n",
+    "  * **Pre-training Bias**: Pre-training bias metrics to be computed. The detailed description of the metrics can be found on [Measure Pretraining Bias](https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-measure-data-bias.html). This example sets methods to \"all\" to compute all the pre-training bias metrics.\n",
+    "  * **Post-training Bias**: Post-training bias metrics to be computed. The detailed description of the metrics can be found on [Measure Posttraining Bias](https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-detect-post-training-bias.html). This example sets methods to \"all\" to compute all the post-training bias metrics.\n",
```

Comment:
Major comment: similarly configuration key followed by its description then change `Post-training Bias` to the key `post_training_bias`,

* `post_training_bias`: Post-training bias metrics to be computed. The detailed description of the metrics can be found on [Measure Posttraining Bias](https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-measure-post-training-bias.html). This example sets methods to "all" to compute all the post-training bias metrics.